### PR TITLE
Update CHANGELOG.md for 2.5.0

### PR DESCRIPTION
**Pull request by jforissier**
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,125 @@
...
+
+* New supported platform: i.MX7D ([#1639])
+* Secure storage: anti-rollback protection for REE FS using RPMB FS ([#1630])
+* Assign non-secure DDR configuration from DT if CFG_DT=y ([#1623])
+* Add new image format: split image into three separate binaries suitable for
+  upcoming ARM Trusted Firmware ([#1589]).
+* Make alignment check configurable ([#1586])
+* drivers: add TZC380 driver ([#1578])
+* plat-imx: PSCI CPU off ([#1577])
+* 64-bit paging on QEMU v8 and HiKey ([#1575], [#1592])
```
Comment (jenswi-linaro) [inline CHANGELOG.md]: Juno and FVP too

**Comment by sorenb-xlnx** [comment]
Things look good on zynqmp:
```
47112 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```

**Comment by jenswi-linaro** [comment]
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> Juno AArch{32,64}`
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> Juno AArch{32,64} pager`

**Comment by glneo** [comment]
Tested-by: Andrew F. Davis <afd@ti.com>

For:

ti-dra7xx
ti-am57xx
ti-am43xx

**Comment by jenswi-linaro** [comment]
```
Tested-by: Jens Wiklander <jens.wiklander@linaro.org> FVP AArch{32,64}
Tested-by: Jens Wiklander <jens.wiklander@linaro.org> FVP AArch{32,64} pager
```

**Comment by igoropaniuk** [comment]
```
$ xtest -l 15
+-----------------------------------------------------
47112 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```

`Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org> (RPi3)`
Known https://github.com/OP-TEE/optee_os/issues/1353 issue still reproduces

**Comment by jbech-linaro** [comment]
```bash
NOTICE:  BL31: v1.2(debug):8b2410c
NOTICE:  BL31: Built : 09:52:53, Jun 30 2017
...
INFO:    TEE-CORE: OP-TEE version: 2.5.0-rc1 #1 fre 30 jun 2017 07:52:46 UTC aarch64
...
[    0.000000] Linux version 4.9.0-61854-g2e26af9 (jbech@guppy) (gcc version 6.2.1 20161016 (Linaro GCC 6.2-2016.11) ) #1 SMP PREEMPT Fri Jun 30 09:53:20 CEST 2017

root@MT8173_EVB:/ xtest -l 15
....
+-----------------------------------------------------
47112 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```
All fine, but this still uses its own ARM-TF (very old branch). I plan to update to use Liang's new branch after the 2.5.0 release.

`Tested-by: Joakim Bech <joakim.bech@linaro.org>` (MTK8173)

**Comment by jforissier** [comment]
For tag 2.5.0-rc1:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

Aarch32, Aarch64, with and without pager. All test configurations were run with GP tests enabled.
Debian build is OK too.

**Comment by etienne-lms** [comment]
For tag 2.5.0-rc1:
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org>` (b2260, GP)
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org>` (qemu_virt, GP)
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org>` (qemu_armv8, GP)

Tested with-and-without pager for the 3 setups.
Tested with-and-without-SDP for qemu_armv8.

**Comment by etienne-lms** [comment]
Oups, almost forgotten this one:
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (cannes2, GP)

**Comment by b49020** [comment]
Tested-by: Sumit Garg <sumit.garg@nxp.com> (ls1021atwr-single-core)

root@ls1021atwr:~# ./xtest -l 15
Run test suite with level=15
....
+-----------------------------------------------------
47118 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!

**Comment by lorc** [comment]
`Tested-by: Volodymyr Babchuk <vlad.babchuk@gmail.com> (RCAR H3)`

> 47112 subtests of which 0 failed
> 78 test cases of which 0 failed
> 0 test case was skipped
> TEE test application done!
>

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (D02)`

**Comment by vchong** [comment]
For tag 2.5.0-rc1:

`Tested by: Victor Chong <victor.chong@linaro.org> (hikey aosp)`

```
23905 subtests of which 0 failed                                                                                       
79 test cases of which 0 failed                                                                                        
0 test case was skipped  
TEE test application done!                                                                                             
root@hikey:/ # 
```

**Comment by MrVan** [comment]
only test with xtest
```
Tested-by: Peng Fan <peng.fan@nxp.com> (imx-mx7dsabresd)
Tested-by: Peng Fan <peng.fan@nxp.com> (imx-mx6ulevk)
Tested-by: Peng Fan <peng.fan@nxp.com> (imx-mx6ullevk)
```

**Comment by jbech-linaro** [comment]
> @etienne-lms  Tested with-and-without-SDP for qemu_armv8.

Did you have any issues with libfdt on QEMU v8? I'm running `Ubuntu 16.04.2 LTS` and with that I get libfdt `1.4.0+dfsg-2`. I.e., it seems like I need to manually update it to latest libfdt like we are doing in [Travis](https://github.com/OP-TEE/build/blob/master/.travis.yml#L56-L57), since otherwise QEMU complains that it needs at least libfdt `1.4.2`. I.e., if I'm not the only one seeing this, it looks like our manifest/build combination doesn't work for QEMU v8. @jenswi-linaro you also tested v8 according to the comments above. You didn't face the issue?

It would be interesting to know if you get newer libfdt on new Ubuntu versions. Eventually I can try that in Docker.

**Comment by jenswi-linaro** [comment]
@jbech-linaro no I haven't tested qemu yet. I just did a test compile and I'm also facing libfdt issues when compiling qemu.

**Comment by etienne-lms** [comment]
@jbech-linaro: i run an Ubuntu 14.04 LTS and have the libfdt 1.4.0 installed: qemu is fine.
I checked my setup: indeed, I ran the new optee_* 2.5.0-rc1 above an old qemu reference.
I am fetching the full brand new 2.5.0-rc1 manifest and will tell you.

**Comment by jbech-linaro** [comment]
FYI, with Ubuntu 17.04 you will still get the 1.4.0 version (I just checked using Docker). I wonder why they are using such an old version, it's from 2013 ([releases](https://github.com/dgibson/dtc/releases)). To me it seems like we should update the Makefile in build.git for QEMU_v8, to "manually" checkout the needed version as required by QEMU itself.

**Comment by etienne-lms** [comment]
libfdt issue: proposed https://github.com/OP-TEE/build/pull/156 from Joakim suggestion. Works nice in my env.

**Comment by etienne-lms** [comment]
I removed my `Tested-by:` tag for the 2 qemu targets as the test I made were based on a older qemu package. This test must be launch from scratch (from fetch the full setup, building and testing).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,138 @@
...
+
+* armv7: some platform-specific code (`plat_cpu_reset_early()`) overwrites
+SCTLR bits configured by generic code. This affects alignment checks (`SCTLR.A`)
+and write-implies-no-exec (`SCTLR.WXN`, `SCTLR.UWXN`), which can therefore not
+be configured via the compile-time `CFG_` variables.
+* armv7: plat-imx: Cortex-A9 cores should enable branch prediction (`SCLTR.Z`)
+for improved performance.
+* [#1656] qemu_armv8a: init hangs when secure data path and pager are both
+  enabled.
+
```
Comment (igoropaniuk) [inline CHANGELOG.md]: For RPi3 still actual https://github.com/OP-TEE/optee_os/issues/1353

Comment (jforissier) [inline CHANGELOG.md]: I prefer not to repeat older known problems. In other words, only mention things that have changed compared to the previous release. Perhaps "Known issues" should be renamed to "New issues"?

**Comment by etienne-lms** [comment]
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org>` (qemu_virt)
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org>` (qemu_armv8)

Both tested with pager on and off.
Tested with GP testsuite (when pager off).

**Comment by jforissier** [comment]
For 2.5.0-rc2:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by sorenb-xlnx** [comment]
For 2.5.0-rc2:
```
47112 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```
`Tested-by: Sören Brinkmann <soren.brinkmann@xilinx.com> (zynqmp)`

**Comment by jforissier** [comment]
@etienne-lms the qemu_v8 build fails for me with a "core dump" when building `edk2`. Similar to https://travis-ci.org/OP-TEE/build/jobs/250245963#L5042 actually. Have you seen this already?
```
repo init -u https://github.com/OP-TEE/manifest.git -m qemu_v8_stable.xml -b 2.5.0-rc2
repo sync -d
cd build
make toolchains
make
```

**Comment by etienne-lms** [comment]
Ooh, I fetch the latest master for all manifests I tested, not the 2.5.0-rc2 branch.
I will test again all these from scratch tonight.

Hence, once again, i remove by `Tested-by:` tag for the qemu :(

**Comment by jbech-linaro** [comment]
I think I've seen similar problems, maybe a bit different, in my case I
think was missing "iasl" package on my system.

//Joakim

Den 5 juli 2017 6:36 em skrev "Jérôme Forissier" <notifications@github.com>:

@etienne-lms <https://github.com/etienne-lms> the qemu_v8 build fails for
me with a "core dump" when building edk2. Similar to
https://travis-ci.org/OP-TEE/build/jobs/250245963#L5042 actually. Have you
seen this already?

repo init -u https://github.com/OP-TEE/manifest.git -m
qemu_v8_stable.xml -b 2.5.0-rc2
repo sync -d
cd build
make toolchains
make

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/OP-TEE/optee_os/pull/1649#issuecomment-313157391>, or mute
the thread

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,138 @@
...
+If a platform is not listed, it means the release was not tested on this
+platform.
+
+<!-- ${PLATFORM}-${PLATFORM_FLAVOR}, ordered alphabetically -->
+* d02: TBD
+* hikey: TBD
+* imx-mx6dlsabresd: TBD
+* imx-mx6qsabrelite: TBD
+* imx-mx6qsabresd: TBD
```
Comment (MrVan) [inline CHANGELOG.md]: imx-mx7dsabresd imx-mx6ulevk are also supported.

**Comment by etienne-lms** [comment]
@jforissier : yes same kind of issue.

I followed your command set over the qemu again.
- default: no issue so far (fetch/build/run ok defcfg, with pager, kaddr-sanitize on going)
- default_stable:  (fetch/build/run ok defcfg, with pager, kaddr-sanitize on issue on regression_6016.1)
- qemu_v8: not done
- qemu_v8_stable: build issue on edk2. Faulty commit is the ref.
  Test against commit right below head proposal on edk2 (checkout f7bd152c2a05 "BaseTools: Update tools_def.template") (found from bisect)
 => fetch/build/run ok (defcfg, pager)
 Can you update qemu_v8_stable.xml with this edk2 reference?

**Comment by etienne-lms** [comment]
To fix qemu_v8_stable for 2.5.0 and fix qem_v8 until edk2 issue is fixed:
```patch
--- a/qemu_v8.xml
+++ b/qemu_v8.xml
@@ -35,7 +35,8 @@
        <!-- We're using Linaro SWG for ARM-TF until merged officially -->
        <project remote="linaro-swg" path="arm-trusted-firmware" name="arm-trusted-firmware.git" revision="refs/heads/optee_v2.1.0_paged_armtf_v1.2" />
 
-        <!-- Tianocore, EDK2 -->
+        <!-- Tianocore, EDK2 (issue from e4129b0e5897d76885170bec9da996b266f185f9) -->
+        <project name="edk2.git" path="edk2" remote="tianocore" revision="f7bd152c2a05bd75471305184c25f14f01ccf0b7"/>
         <project remote="tianocore" path="edk2" name="edk2.git" />
 
        <!-- strace -->
--- a/qemu_v8_stable.xml
+++ b/qemu_v8_stable.xml
@@ -16,7 +16,7 @@
   <project name="busybox.git" path="busybox" remote="busybox" revision="dbf5a6da6a4295ce26edd1ce34fde567d19afa02"/>
   <project name="code" path="strace" remote="sfnet" revision="c136e9e449fe733e77e8766a9b26e6dc1134203e"/>
   <project name="dtc.git" path="qemu/dtc" remote="qemu" revision="d990b8013889b816ec054c7e07a77db59c56c400"/>
-  <project name="edk2.git" path="edk2" remote="tianocore" revision="e4129b0e5897d76885170bec9da996b266f185f9"/>
+  <project name="edk2.git" path="edk2" remote="tianocore" revision="f7bd152c2a05bd75471305184c25f14f01ccf0b7"/>  
   <project name="gen_rootfs.git" path="gen_rootfs" remote="linaro-swg" revision="cb8c7736a46d1ff459f159b2115395d504c96b45"/>
   <project name="hello_world.git" path="hello_world" remote="linaro-swg" revision="2c15acb431bcde20beb0277c3acdab6d0b4894ef"/>
   <project name="linux.git" path="linux" remote="linaro-swg" revision="cc2e97bbe2ed6290df25087d8967429071abb214"/>
```

**Comment by jforissier** [comment]
@etienne-lms thanks for investigating. https://github.com/OP-TEE/manifest/pull/62
`qemu_v8_stable.xml` will be updated when the next tag is created (2.5.0, hopefully).

**Comment by jbech-linaro** [comment]
> To fix qemu_v8_stable for 2.5.0 and fix qem_v8 until edk2 issue is fixed:

Good that you found it, what are the odds that it was introduced almost at the same time as we were about to make the release (the one causing the issue was merged 26 hours ago). Luck is not on our side this time :)

**Comment by vchong** [comment]
For tag 2.5.0-rc2:

`Tested by: Victor Chong <victor.chong@linaro.org> (hikey aosp)`

```
23906 subtests of which 0 failed                                                                                       
79 test cases of which 0 failed                                                                                        
0 test case was skipped                                                                                                
TEE test application done!                                                                                             
root@hikey:/ #
```

**Comment by jbech-linaro** [comment]
For tag 2.5.0-rc2:
With https://github.com/OP-TEE/manifest/pull/62 applied on QEMUv8 (only)
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (Juno, QEMUv7, QEMUv8, FVP, MTK8173)`

**Comment by jforissier** [comment]
For tag 2.5.0-rc2:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (D02)`

**Comment by jforissier** [comment]
The HiKey Debian build (https://github.com/OP-TEE/manifest/blob/2.5.0-rc2/hikey_debian_stable.xml) fails for me:
```
Downloading Debian root fs ...
wget http://builds.96boards.org/snapshots/reference-platform/debian-iot/latest/hikey/hikey-rootfs-debian-stretch-iot-20170331-56.emmc.img.gz -O /home/jerome/work/optee_repo_hikey_debian/build/../out/debian_system.img.gz
--2017-07-06 14:16:34--  http://builds.96boards.org/snapshots/reference-platform/debian-iot/latest/hikey/hikey-rootfs-debian-stretch-iot-20170331-56.emmc.img.gz
Resolving builds.96boards.org (builds.96boards.org)... 54.205.27.170
Connecting to builds.96boards.org (builds.96boards.org)|54.205.27.170|:80... connected.
HTTP request sent, awaiting response... 404 NOT FOUND
2017-07-06 14:16:35 ERROR 404: NOT FOUND.

Makefile:305: recipe for target 'system-img' failed
make: *** [system-img] Error 8
```

@vchong any idea which URL should be used? (anything with "latest" **and** a date/version number looks suspicious...)

**Comment by igoropaniuk** [comment]
For tag 2.5.0-rc2:
```
root@RPi3:/bin ./xtest -l 15
+-----------------------------------------------------
47112 subtests of which 0 failed
78 test cases of which 0 failed
0 test case was skipped
TEE test application done!

```
```
Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org> (RPi3)
```

**Comment by etienne-lms** [comment]
For tag 2.5.0-rc2:
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (b2260, QEMUv7)`

**Comment by lorc** [comment]
For tag 2.5.0-rc2:

`Tested-by: Volodymyr Babchuk <vlad.babchuk@gmail.com> (RCAR H3)`

> +-----------------------------------------------------
> 47112 subtests of which 0 failed
> 78 test cases of which 0 failed
> 0 test case was skipped
> TEE test application done!
>

**Comment by MrVan** [comment]
For tag 2.5.0-rc2:
Tested with running ./xtest
`Tested-by: Peng Fan <peng.fan@nxp.com> (imx-mx7dsabresd imx-mx6ulevk imx-mx6ullevk)`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,130 @@
+# OP-TEE - version 2.5.0-rc1
```
Comment (jbech-linaro) [inline CHANGELOG.md]: Don't forget to change this later when doing the merge.

Comment (jforissier) [inline CHANGELOG.md]: Oops, yes thank you

**Comment by etienne-lms** [comment]
minor: @jforissier, few T-b tag are missing some diamonds around mail address in the commit message you are forging.

**Comment by jforissier** [comment]
@etienne-lms right, will fix.

**Comment by jforissier** [comment]
Merged. Thanks everyone for testing!

**Pull request merged**